src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -346,5 +353,52 @@ private BoundExpression RewriteStringConcatInExpressionLambda(CSharpSyntaxNode s
 
             return new BoundBinaryOperator(syntax, operatorKind, loweredLeft, loweredRight, default(ConstantValue), method, default(LookupResultKind), type);
         }
+
+        /// <summary>
+        /// Checks whether the expression represents a boxing conversion of a value type.
+        /// If it does, it returns a synthesized ToString call for that value type instance.
+        /// Otherwise, it returns the original expression.
+        /// </summary>
+        private BoundExpression ConvertExpressionToStringIfValueType(CSharpSyntaxNode syntax, BoundExpression expr)
+        {
+            // The lowered expressions coming in will be boxing conversions if the input is a value type.
+            // When passed to String.Concat, the resulting object will have ToString called on it.
+            // We can optimize away the boxing allocation by just calling ToString on the value type directly.
```

Comment:
@VladimirReshetnikov do you consider it a breaking change to call ToString on the immutable value types?  Or are you just referring to the current behavior which would call it on any struct?